### DISCUSSION TITLE
Postgres database replica set up

### DISCUSSION THREAD

**Discussion by nicolasdalbroi** [digitalocean/terraform-provider-digitalocean#638] [Q&A]
Hello,

I'm experiencing some issues trying to figure out how to import the database replica to terraform.
When I get the database information with doctl databases list it show all the database except for the databases' replicas. The thing is that when I import the project along with all the resources assigned to it, it shows the replica as a cluster and not a replica.

Should I import the resource as a replica cluster like it should be or as a cluster?

**Comment by andrewsomething** [comment]
Replicas are a sub-resource of the database cluster. With `doctl`, after finding the ID of the parent cluster, you can then list any replicas it has. E.g.

```
$ doctl databases replica list 342e9a37-0913-4c73-8baf-13876c28b7bb
Name                                        Region    Status
db-postgresql-nyc1-32395-read-only-39242    nyc1      online
```

To import a [replica into Terraform](https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/resources/database_replica#import),  you need to provide the ID of the parent database cluster and the name of the replica joined with a comma. E.g.

```
terraform import digitalocean_database_replica.read-replica 342e9a37-0913-4c73-8baf-13876c28b7bb,db-postgresql-nyc1-32395-read-only-39242
```

**Comment by nicolasdalbroi** [comment]
I understand, but when I import the project I get the replica urn, but when I try to update it the urn argument it's not valid. Should I use the id to import the replica into the project?

**Reply by holms** [reply]
I believe you need to provide exact steps of what you're trying to do.
